proof/refine/ARM/Tcb_R.thy

```diff
@@ -3143,6 +3143,13 @@ lemma checkPrio_lt_ct_weak:
   apply (clarsimp simp: pred_tcb_at'_def obj_at'_def)
   by (rule le_ucast_ucast_le) simp
 
+lemma checkPrio_lt_ct_weak':
+  "\<lbrace>P\<rbrace> checkPrio prio auth \<lbrace>\<lambda>rv s. P s \<and> mcpriority_tcb_at' (\<lambda>mcp. ucast prio \<le> mcp) auth s\<rbrace>, -"
```

Comment:
Ooh, that's a nice way of solving the problem we discussed. I guess that means that the idea of replacing `checkPrio_inv` with several more specific lemmas didn't work though?

Reply:
Yeah that quickly became annoying. The `wp` produced some postconditions where the preemption case was not simply `True`, and I thought the approach to fixing it was worse than the slightly more manual forwards proof. This proof does take quite a while to get through the case spliting in the first few lines, but I think it's much clearer what's going on